### ISSUE TITLE
Can't use k of 39 (must be <= 31)

### ISSUE THREAD

**Issue by ailtonpcf** [fbreitwieser/krakenuniq#160]
Hey,
I would like to try Krakenuniq with kmers longer than 31, however I get an error: 
`Can't use k of 39 (must be <= 31)` 

Is that really true that I can't use bigger kmer values, or a bug?

Thanks in advance,
Ailton.

**Comment by salzberg** [comment]
yes that's correct - Krakenuniq uses an array that can only hold values up to 2^31 and there are places in the code that would break if you used longer k-mers. (And yes, I know that there are 4^31 possible DNA sequences of length 31.) Changing this would require a major code rewrite. Btw with longer k-mers you lose sensitivity rather quickly.
